### ISSUE TITLE
`BLOCKED`: How do I denote the "urgency" and "importance" of an issue beyond "priority-1"?

### ISSUE THREAD

**Issue by nelsonic** [dwyl/labels#87]
# Context

Last month I opened an issue https://github.com/dwyl/cid/issues/11 that is a "**Blocker**" to the progress of ***all*** other work 
(_both our Product(s) and Client work_). _We_ ***MUST*** get it done as the _highest_ priority and the longer we go on without it being complete, the more "re-work" we will have to do in future. This is basically a ["**cash _bonfire_**"](https://youtu.be/PYuzqNSPDsI) we need to _resolve_ and the longer we ignore it, the more money it is costing us. 💸 🔥 

![image](https://user-images.githubusercontent.com/194400/51378500-4d77e400-1b05-11e9-8c6d-542284dd34d1.png)

> **To be _clear_** `this` issue is not about _money_, but if we lose sight of what issues are ***costing*** us cash every _minute_ that passes, we will _literally_ go bankrupt", the dream of "dwyl" ends and everyone will have to find another job. 😞 If we put out this "fire" we can save some of our "runway". 

# Question

### So my/the Question is: how do I/we make it _clear_ to _everyone_ in the team that **_NOTHING_ `else`** is more important than getting `this` "_one thing_" `done`?

How do we _systematically_ communicate with everyone in the @dwyl "core" team and wider community (_where applicable_) that getting a _specific_ issue "unblocked" is the difference between success and failure of the organisation/company? 

# Proposal: `BLOCKED` Label

I propose that we start by creating a new label called **`BLOCKED`** 
(_intentional ALLCAPS as the person applying the label is "screaming for help!"_)

This label should only be applied when the person has exhausted all other options and has made a concerted effort to do everything they can to describe the problem they are facing. 

When the **`BLOCKED`** label is applied it should trigger an **`alert`** to call _everyone's_ attention.
***everyone*** in the "core" team should attempt to help in solving the issue. 
If a corresponding StackOverflow issue is opened for it, upvote it so it gains exposure.

# Wider Discussion

GitHub is _definitely not_ a _good_ "company" management system. (_this is not news to anyone_)
GitHub is _fine_ for tracking the issues of a software project but it's inadequate for tracking development across multiple organisations for example when we have a dependency **`fields`** that need to be updated before the client feature can be completed, it's difficult to visualise this dependency in the "milestone" or "project board". 

The _only_ reason we started using GitHub to collect the information/stories/tasks/discussions
for @dwyl was because our "core" product is code/software and GitHub is the most _popular_ platform (_by far_) for collaborating on open source code. I _still_ feel that it's (_way_) more important to have a single source of truth and avoid fragmenting project information across multiple communication systems.

## `Priority-0` ?

@iteles and I have _informally_ (_verbally_) discussed the _idea_ of having a **`priority-0`** label.
The _reason_ we have _resisted_ doing this is simple: the _escalation_ to **`p0`** would simply "bump" the whole priority system up a level and we would end up with 10 **`p0`** items and back in the same place we are now ... 🙄 (_also, the irony of the lack of transparency of a verbal conversation is not lost on me, despite my insistence that all discussions should be had on GitHub for traceability/accountability, many still happen on Gitter or in-person and never relayed onto our "single-source-of-truth" ..._)

## Dependency Visualisation

We need a _single_ dashboard for visualising what everyone in the team is _currently_ working on, what they will be working on _next_ and what issues are blocked across all teams/orgs/clients/projects.

The longer we leave this the more likely we are to run out of cash.

## Longer Term 

In the _near_ future we will have a [Gantt Chart](https://en.wikipedia.org/wiki/Gantt_chart) https://github.com/dwyl/product-roadmap/issues/8 to help us _visualise_ the order/priority of issues/stories/tasks so that it's _immediately_ clear to everyone what needs to be done _now_, _next_ or _never_. (_some things are so low on the priority list - because users aren't requesting them and they will deliver little value while cluttering the UI so - they should simply **never** get done if we are prioritising correctly, and yet **often** they get "actioned" because of a lack of clarity... there is **way** to much at stake for us to lose focus and do low priority tasks!_)

**Comment by iteles** [comment]
Linked to https://github.com/dwyl/hq/issues/349

My hypothesis is that`BLOCKED` may be used across the board by anyone who is blocked in their own project and will quickly lose impact.   
In an open source focused organisation like ours, anyone who has a question or has their own work blocked when using one of our open source modules is likely to use it. 

Using something like `dwyl-priority`, `priority-0` or `priority-nuclear` would denote a bit more seniority and disavow people from the notion that it can just be used at will. 

It may also be advantageous to start off with the rule that organisation-wide, there can only be _one_ `priority-0` (or whatever we call it) at any given time, until we feel there is an absolute ***need*** to expand this number. This can be written into the label description so it's obvious immediately at the point of applying the label.

**Comment by iteles** [comment]
Created a `BLOCKED` label given the priority here, name can be changed if this isn't working later on.
![image](https://user-images.githubusercontent.com/4185328/51414579-e4c35280-1b6a-11e9-9005-a353c8f340ae.png)

**Comment by nelsonic** [comment]
Inês, this is a _valid_ hypothesis. 🤔 

_However_ only people who are _members_ of the organisation are _able_ to add labels so we would not have random chancers using our [Google Sheets Tutorial](https://github.com/dwyl/learn-to-send-email-via-google-script-html-no-server) which often has Noobs making "do my work for free" requests. 

If someone who _is_ a collaborator adds the **`BLOCKED`** label we will have @dwylbot comment linking to our "workflow" which will explain the _exact_ circumstances when the label can be applied.
The _ONLY_ circumstances when it *Can* are:

1. @dwyl the company (_or one of our customers_) is losing money while the issue remains unfixed.
2. Development of a @dwyl product/service or client app is "blocked" until the issue is fixed
3. There is a security issue, though I feel we should have a _separate_ "channel" for ["responsible disclosure"](https://en.wikipedia.org/wiki/Responsible_disclosure) of security related issues rather than people posting public GitHub issues ... 

The @dwylbot comment will give the person a few minutes to re-think if they "_crying wolf_". 
After that "cooling off" time, @dwylbot should email both Inês and Nelson (_and anyone else in the intersection of the "contributor" list and "core" team_) 

I can think of fewer than 10 instances in the (4 year) history of @dwyl where I would have applied this label.

**`priority-0`** does not give anyone any indication that the whole company is Blocked until the issue is resolved and again, it will almost inevitably result in a "priority escalation".

At present there are 56 open **`priority-1`** issues:
https://github.com/issues?q=is%3Aissue+is%3Aopen+label%3Apriority-1+user%3Adwyl
![image](https://user-images.githubusercontent.com/194400/51414137-41257280-1b69-11e9-8428-b65f1d48e8e1.png)

This is a _clear_ indication that:
A. we are not _systematically_ working on the most important issues/tasks.
and/or
B. the priority system is not clearly _understood_ by the team.

Of these issues _zero_ are assigned to me. 
https://github.com/issues?q=is%3Aissue+is%3Aopen+label%3Apriority-1+user%3Adwyl+assignee%3Anelsonic
![image](https://user-images.githubusercontent.com/194400/51414181-60240480-1b69-11e9-9bf2-d0076c573f9c.png)

But if a **`BLOCKED`** label was applied to anything I would know that regardless of who an issue/task is assigned to, I need to look at it. 
Whereas if a **`priority-0`** is applied but someone is assigned I _assume_ that the person is _actively_ working on the issue ... or is that a _bad_ assumption unless it has **`in-progress`**?

All of this would be a _lot_ easier for _everyone_ to understand if the "workflow" was clarified /github.com/dwyl/contributing/issues/109

**Comment by nelsonic** [comment]
@iteles does the assignment to me signify _agreement_ with this label suggestion/idea?
If so, should we _document_ it's usage guidelines?

**Comment by iteles** [comment]
@nelsonic The label has been available and in use since the beginning of the year https://github.com/dwyl/labels/issues/87#issuecomment-455698834

Is there anything else you'd like to add to the above ahead of it being documented?

**Comment by nelsonic** [comment]
@iteles if you feel the above is sufficiently clear, then no (_for the time being_).
